sootup.java.bytecode/src/main/java/sootup/java/bytecode/frontend/AsmMethodSource.java

```diff
@@ -243,15 +212,25 @@ public Body resolveBody(@Nonnull Iterable<Modifier> modifierIt) {
     operandStack = null;
 
     bodyBuilder.setMethodSignature(lazyMethodSignature.get());
-
-    for (BodyInterceptor bodyInterceptor : view.getBodyInterceptors()) {
-      try {
-        bodyInterceptor.interceptBody(bodyBuilder, view);
-      } catch (Exception e) {
-        throw new IllegalStateException(
-            "Failed to apply " + bodyInterceptor + " to " + lazyMethodSignature.get(), e);
+    List<AnalysisInputLocation> inputLocations = view.getProject().getInputLocations();
+    Optional<AnalysisInputLocation> containingInputLocation =
+        inputLocations.stream()
+            .filter(i -> i.getClassSource(this.declaringClass, view).isPresent())
+            .findFirst();
+    // we expect the class to come from a single input location
```

Comment:
field view is now not used anymore - we could replace it in the ctor/single usage in asmclassource by the inputlocation to avoid the search of inputlocations